src/osd/ECCache.hpp

```diff
@@ -0,0 +1,273 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2016 Western Digital Corporation.
+ * Author: Tomy Cheru <tomy.cheru@sandisk.com>
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#ifndef ECCACHE_HPP_INCLUDED
+#define ECCACHE_HPP_INCLUDED
+
+#include <iostream>
+//#include <string>
+#include <set>
+#include "include/buffer.h"
+#include "common/hobject.h"
+#include "common/Mutex.h"
+#include "common/Cond.h"
+
+class ecc_extent
+{
+  private:
+    unsigned eoff;
+    bufferlist edata;
+
+  public:
```

Comment:
In the ceph code base, header files (almost) always have the open brace on the same line as the declaration as long as it doesn't end up longer than 80 characters, please fix that.